File: src/main/resources/output/IFC4X3_js/Types/IfcRotationalStiffnessSelect.js
// SELECT TYPE IfcRotationalStiffnessSelect
// Options:
// - IfcBoolean
// - IfcRotationalStiffnessMeasure
class IfcRotationalStiffnessSelect {
    constructor(value) {
        if (!(value instanceof IfcBoolean || value instanceof IfcRotationalStiffnessMeasure)) {
            throw new Error("Value must be one of: IfcBoolean, IfcRotationalStiffnessMeasure");
        }
        this.value = value;
    }

    getValueAs(TypeClass) {
        if (this.value instanceof TypeClass) {
            return this.value;
        } else {
            throw new Error("Stored value is not of the requested type");
        }
    }
}

File: src/main/resources/output/IFC4X3_java/Types/IfcRotationalStiffnessSelect.java
// SELECT TYPE IfcRotationalStiffnessSelect
public class IfcRotationalStiffnessSelect {
    public Object value;

    public IfcRotationalStiffnessSelect(Object value) {
        if (!(value instanceof IfcBoolean || value instanceof IfcRotationalStiffnessMeasure)) {
            throw new IllegalArgumentException("Value must be one of: IfcBoolean, IfcRotationalStiffnessMeasure");
        }
        this.value = value;
    }

    public <T> T getValueAs(Class<T> clazz) {
        return clazz.cast(value);
    }
}
